docs/source/cloud-setup/cloud-permissions/gcp.rst

```diff
@@ -1,19 +1,56 @@
-.. _cloud-permissions-gcp:
-
 GCP
 =============
 
 
+.. _gcp-service-account:
+
+GCP service account
+-------------------
+
+`GCP service accounts
+<https://cloud.google.com/iam/docs/service-account-overview>`__ are supported.
+
+.. tip::
+    A service account on your local machine can avoid the periodic
+    ``google.auth.exceptions.RefreshError: Reauthentication is needed. Please
+    run `gcloud auth application-default login` to reauthenticate.`` error. A
+    service account is long-lived as it does not have an expiry time.
+
+Set up a service account as follows:
+
+1. Follow the :ref:`instructions <gcp-service-account-creation>` to create a service account with the appropriate roles/permissions.
+2. In the "Service Accounts" tab in the `IAM & Admin console
+   <https://console.cloud.google.com/iam-admin/iam>`__, click on the service
+   account to go to its detailed page. Click on the **KEYS** tab, then click on
+   **ADD KEY** to add a JSON key.  The key will be downloaded automatically.
+3. Set the environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` to the path of the key file, and configure the gcloud CLI tool:
+
+   .. code-block:: console
+
+       $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/key.json
+       $ gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+       $ gcloud config set project your-project-id
+
+   You may want to add the export statement in your profile (e.g. ``~/.bashrc``, ``~/.zshrc``) so that it is set automatically in all new terminal sessions.
+
```

Comment:
We should also check when/why we need `gcloud auth login` and `gcloud auth application-default login`. It is currently quite unclear in the doc and always confuse me.

Reply:
To be clear, this is copied from the deleted file and I didn't change the content. But yes - you can just do `gcloud auth login --update-adc`. We should update this, but maybe not in this PR.

Reply:
Sounds good. Just want to make sure this is on our radar. IIRC, the former is required for GCS access while the latter is needed for creating VMs.

Reply:
Also, we should check if the service account credentials get uploaded correctly.

Reply:
> Also, we should check if the service account credentials get uploaded correctly.

@Michaelvll Looks like it should: https://github.com/skypilot-org/skypilot/blob/e536958f1e9cd2bf91e379bd8381dbb8c084c0b1/sky/clouds/gcp.py#L898-L902 It will check the `GOOGLE_APPLICATION_CREDENTIALS` env var.

I will try it though to double check.

Reply:
Uncovered #5210, but otherwise it works.